### ISSUE TITLE
Declaration of Properties.Resources.Culture is invalid for nullable projects

### ISSUE THREAD

**Issue by MelonWang1** [dotnet/project-system#9196]
## Visual Studio Version
17.8.0 Preview 1.0 [33927.345.main]

## Steps to Reproduce

1. Create a C# .NET Core WinForms core application
2. Ctrl+Shift+A to add a resources file
3. Go to the Resources.Designer.cs file to observe

## Expected Behavior
The above mentioned code snippet rather should be looking like this:
![image](https://github.com/dotnet/project-system/assets/94418985/38304745-7f9e-4c16-ae9e-364dba480472)

## Actual Behavior
The Resources.Designer.cs file generated by Visual Studio 2022 contains the following code snippet:
![image](https://github.com/dotnet/project-system/assets/94418985/1f388c96-d6f6-4efc-98dc-7dc5caa57911)

##More Info
1. The initial ticket link is here: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1781240.

**Comment by drewnoakes** [comment]
The generation of this code is not owned by this repo. The editor UI is here, but that just edits the `.resx` file. A separate `ResXFileCodeGenerator` custom tool generates the `.Designer.cs` file.

I believe the generation is performed using CodeDom which is an older API and does not support Nullable Reference Types.

These days the use of VS [custom tools](https://learn.microsoft.com/en-us/visualstudio/extensibility/internals/custom-tools?view=vs-2022) is discouraged, as they only work inside Visual Studio. A better approach might be to use a source generator. The dotnet/roslyn-analyzers repo has one (https://github.com/dotnet/roslyn-analyzers/tree/main/src/Microsoft.CodeAnalysis.ResxSourceGenerator) but I'm not sure what it's status is or if it ships anywhere. cc @sharwell

**Comment by sharwell** [comment]
Yes, ResxSourceGenerator should be usable now. I think @AArnott posted an article about how to start using it.

**Comment by sharwell** [comment]
Note that nullable reference types are disabled in the default Resources.Designer.cs file. Attempting to declare the property as nullable without other changes would result in warnings about the use of nullability annotations in a nullable disabled context.

**Comment by AArnott** [comment]
Here is my blog post on the migration: https://blog.nerdbank.net/2023/04/get-rid-of-your-designer-cs-code-behind-for-resx

**Comment by smitpatel** [comment]
We need to figure what exactly to do.

**Comment by drewnoakes** [comment]
I spoke with @merriemcgaw and @Tanya-Solyanik and they own the generator here, so moving this to dotnet/winforms.
